api/src/main/java/org/apache/iceberg/io/FileIO.java

```diff
@@ -68,6 +68,15 @@ default void deleteFile(OutputFile file) {
   default void initialize(Map<String, String> properties) {
   }
 
+  /**
+   * Delete the files at the given path.
+   */
+  default void deleteFiles(Iterable<String> pathsToDelete) {
```

Comment:
I'm wondering if it makes more sense to just have this interface take a set of paths to delete. That way we don't have to worry about de-duping ourselves. I would imagine this is something that would be common across all FileIO.

Reply:
I guess it's fine to leave it as Iterable as I feel the java set also implements the Iterable interface, so if needed the consumer of this method can pass the set for de-duplication purpose. If there's a need to preserve the ordering, then they can pass other iterable like list as well. From correctness perspective, the concrete file system shall be handle the duplicate file path for deletion.

Reply:
Would S3 API fail us if there is any duplicated path in the input request? Have you tried out?

Reply:
For both duplicated paths and in the case the key does not exist (already deleted) S3 will not fail.